public/app/plugins/datasource/prometheus/configuration/Connection.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState, useCallback } from 'react';
+
+import { DataSourceSettings } from '@grafana/data';
+import { selectors } from '@grafana/e2e-selectors';
+import { InlineField, Input, useTheme2 } from '@grafana/ui';
+import { HttpSettingsBaseProps } from '@grafana/ui/src/components/DataSourceSettings/types';
+
+import { docsTip, overhaulStyles, PROM_CONFIG_LABEL_WIDTH, validateInput } from './ConfigEditor';
+
+export interface ConnectionProps extends HttpSettingsBaseProps {
+  /** The default url for the data source */
+  defaultUrl?: string;
+}
+
+export const Connection = (props: ConnectionProps) => {
+  const { defaultUrl, dataSourceConfig, onChange } = props;
+
+  const theme = useTheme2();
+  const styles = overhaulStyles(theme);
+
+  const [validPromUrl, updateValidPromUrl] = useState<string>('');
+
+  const onSettingsChange = useCallback(
+    // eslint-disable-next-line
+    (change: Partial<DataSourceSettings<any, any>>) => {
+      onChange({
+        ...dataSourceConfig,
+        ...change,
+      });
+    },
+    [dataSourceConfig, onChange]
+  );
+
+  let urlTooltip;
+
+  switch (dataSourceConfig.access) {
+    case 'direct':
+      urlTooltip = (
+        <>
+          Your access method is <em>Browser</em>, this means the URL needs to be accessible from the browser.
+          {docsTip()}
+        </>
+      );
+      break;
+    case 'proxy':
+      urlTooltip = <>This URL must be accessible from the Grafana server. {docsTip()}</>;
+      break;
+    default:
+      urlTooltip = 'Specify a complete HTTP URL (for example http://your_server:8080)';
+  }
+
+  const validUrlRegex = /^(ftp|http|https):\/\/(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!\-\/]))?$/;
```

Comment:
Where did we get this from? Is this currently in use in grafana anywhere? Do we have any additional documentation as to what urls this accepts?

Maybe unit tests for this if it's not already tested

Reply:
This connection component logic was copied from https://github.com/grafana/grafana/blob/39a3d85514553e87bd17e516bcf97373e14ad9f6/packages/grafana-ui/src/components/DataSourceSettings/DataSourceHttpSettings.tsx#L122-L124 which implements this regex... but it does not test it 😆 I will write some unit tests for it.

Reply:
tests added! https://github.com/grafana/grafana/pull/66198/commits/7ef949ccefdbaa1bad51b78a717b009baaba6562